requirements.txt

```diff
@@ -6,4 +6,4 @@ packaging
 pandas>=0.23
 xarray>=0.16.1
 netcdf4
-typing_extensions>=3.7.4.3,<4
+typing_extensions>=3.7.4.3
```

Comment:
Consider this as an alternative?
```suggestion
typing_extensions>=3.7.4.3,<5
```

I have no strong feelings one way vs the other — broadly, it's a tradeoff between "prevent possibly-non-working combinations, have to bump version bounds manually" and "users run into non-working combinations and open issues that don't really point to the root cause, setting you up for some fun debugging time".

Reply:
I'm going to leave this as is, since we don't adopt this pinning elsewhere (but helpful to think about the multiple failure modes)